Identity: Added comment about data saved to database (#2915)
camunda/camunda-docs: versioned_docs/version-8.3/self-managed/identity/deployment/configuration-variables.md

### Database configuration

> Identity does require a database to store information about resource-based permissions or multi tenancy configuration.

Reviewer: Suggestion from my side to keep the wording consistent across the docs.

```suggestion
Identity does require a database to store information about resource authorization and multi-tenancy.
```
